engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/hv/UUIDValidator.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.internal.constraintvalidators.hv;
+
+import java.util.Arrays;
+import java.util.Locale;
+
+import org.hibernate.validator.constraints.UUID;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+
+import static org.hibernate.validator.constraints.UUID.LetterCase;
+
+/**
+ * Checks that the annotated character sequence is a valid
+ * <a href="https://en.wikipedia.org/wiki/Universally_unique_identifier">UUID</a>.
+ *
+ * @author Daniel Heid
+ */
+public class UUIDValidator implements ConstraintValidator<UUID, CharSequence> {
+
+	private static final int[] GROUP_LENGTHS = { 8, 4, 4, 4, 12 };
+
+	private boolean allowEmpty;
+
+	private boolean allowNil;
+
+	private int[] version;
+
+	private int[] variant;
+
+	private LetterCase letterCase;
+
+	@Override
+	public void initialize(UUID constraintAnnotation) {
+		allowEmpty = constraintAnnotation.allowEmpty();
+		allowNil = constraintAnnotation.allowNil();
+		version = checkAndSortMultiOptionParameter( constraintAnnotation.version(), "version", 1, 15 );
+		variant = checkAndSortMultiOptionParameter( constraintAnnotation.variant(), "variant", 0, 2 );
+		letterCase = constraintAnnotation.letterCase();
+		if ( letterCase == null ) {
+			throw new IllegalArgumentException( "Letter case must not be null" );
+		}
+	}
+
+	private static int[] checkAndSortMultiOptionParameter(
+			int[] values,
+			String parameterName,
+			int minimum,
+			int maximum) {
+		if ( values == null ) {
+			throw new IllegalArgumentException( String.format(
+					Locale.ROOT,
+					"Parameter %s must not be null",
+					parameterName
+			) );
+		}
+		if ( values.length == 0 ) {
+			throw new IllegalArgumentException( String.format(
+					Locale.ROOT,
+					"Parameter %s must not be empty",
+					parameterName
+			) );
+		}
+		for ( int value : values ) {
+			if ( value < minimum ) {
+				throw new IllegalArgumentException( String.format(
```

Comment:
Could you please update these parameter verification checks to match this style:
```
Contracts.assertTrue(  value < minimum, MESSAGES.parameterShouldBeAtLeastEqualTo( "Minimum version", 1 ) );
```
see here for an example - https://github.com/hibernate/hibernate-validator/blob/main/engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/hv/ParameterScriptAssertValidator.java#L72-L73

Essentially it is the same but would be more inline with the rest of the code. Thanks!

Reply:
You're welcome :sweat_smile: